### DISCUSSION TITLE
GSoC 2024

### DISCUSSION THREAD

**Discussion by Ultiautomation** [alan-turing-institute/sktime#6243] [Q&A] [answered]
Hello @fkiraly, 
Hope you are well. Hi I am Shaibi, a student of Masters in Data Science - set for completion this summer and just discovered this platform through one of my batch mates today. Guess there is a lot to learn here and would definitely be interested in contributing. I was also told that their are projects that I could participate in. It looks though that I am bit too late in the process, so wanted to check in with you if the proposal submission could be still made. I am quiet interested in the Timeseries forecasting area and have done moderate work using Python and supervised ML algorithms earlier and would like to learn more in this area.

If given a chance, I would like to work on the project idea below and I have worked on retail use case for capstone project and hence am familiar with such use case, and concepts applied to categorical variable during data preprocessing. Will await for your advise.  
"categorical and time-constant features in forecasting
extend framework for full support of categorical features and time-constant features
implement transformers and feature extractors, or interface 3rd party
ensure pipelines, tuning, composites, hierarchical support work end-to-end
Skills: sklearn and sklearn extensions with categorical features, framework/python, helpful: familiarity with finance or retail use cases

Getting started: interface a categorical feature encoder

Expected outcomes: basic support for categorical features in sktime

Stretch goal: full end-to-end support, noteobook/tutorials

Mentors: yarnabrina, fkiraly

Difficulty: Medium

Length: 350 hours" 

Thanking you 
Shaibi

**Comment by fkiraly** [comment, marked as answer]
Welcome to `sktime`, @Ultiautomation!

You're not too late for GSoC, the deadline is Apr 2, 18 UTC.
Details on the application process are here: https://github.com/sktime/mentoring/blob/main/internships/gsoc2024.md

There are als various mentoring opportunities, including menotring (unpaid) on a rolling basis, and google season of docs, see here:
https://github.com/sktime/mentoring

**Comment by Ultiautomation** [comment]
@fkiraly <<reporting in right forum >>-  After carefully considering the various opportunities and tasks/issues available within the sktime community, I've come to realize that my forte currently lies more in the realm of documentation and technical writing rather than coding-based challenges. While I'm super excited about being part of sktime, I reckon my skills currently would be better channeled towards the mentorship program where I can take on coding related issues and solve this.

I noticed that there's a project aimed at revamping the documentation for the sktime library, and it seems like there hasn't been much progress since the issue was first opened. Given my background as a seasoned project manager with loads of experience in organizational documentation, I'm pretty confident I could roll up my sleeves and work on this. I've got a knack for crafting content that's clear, concise, and easy to digest.

So, with your permission, I'd like to throw my hat in the ring for GSoD (Google Season of Docs) and help spruce up sktime's documentation. I reckon my skills would mesh well with the project's goals, and I'm super excited to be part of the sktime community in what ever way possible. Do let me know your thoughts.

Thanks a bunch for considering my pitch, and I can't wait to dive into some doc-revamping action with you all!

Cheers,
Shaibi Shamsudeen

**Reply by fkiraly** [reply]
> So, with your permission, I'd like to throw my hat in the ring for GSoD (Google Season of Docs) and help spruce up sktime's documentation.

You don't need my permission to apply, it's open to anyone!
(subject to [google's eligibility criteria](https://developers.google.com/season-of-docs/terms/program-rules))

You can also apply for both GSoC and GSoD, and/or with a shorter GSoC project (90 hours). The metrics module needs some work, so if you found this or #4304 interesting, that could also be a topic.

Work on the docs is of course also very much apprecitaed!
